core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml

```diff
@@ -6568,11 +6606,10 @@ QUALIFY rank_val = (SELECT COUNT(*) FROM emp)]]>
     <Resource name="plan">
       <![CDATA[
 LogicalProject(EMPNO=[$0], ENAME=[$1], DEPTNO=[$2], RANK_VAL=[$3])
-  LogicalFilter(condition=[$5])
-    LogicalProject(EMPNO=[$0], ENAME=[$1], DEPTNO=[$2], RANK_VAL=[$3], EXPR$0=[$4], QualifyExpression=[=($3, $4)])
+  LogicalFilter(condition=[$4])
+    LogicalProject(EMPNO=[$0], ENAME=[$1], DEPTNO=[$7], RANK_VAL=[RANK() OVER (PARTITION BY $1 ORDER BY $7 DESC)], QualifyExpression=[=(RANK() OVER (PARTITION BY $1 ORDER BY $7 DESC), $9)])
```

Comment:
I think the JIRA mentioned something about the ProjectMerge logic in RelBuilder not handling the indexes in the window function. Did that ever get resolved?

Reply:
I don't know. The cause of this bug seems simple - the qualify expression needs to be translated relative to the inputs to the SELECT clause, not relative to the outputs - and when I did that, the query seems to work.

Reply:
Yeah I that makes sense to me